src/bindings/erl_nif/elixir/lib/elektra/system.ex

```diff
@@ -0,0 +1,277 @@
+defmodule Elektra.System do
+  @on_load :load_nifs
+
+  def load_nifs do
+    :erlang.load_nif(Path.join(:code.priv_dir(:elektra), "libnif_kdb"), 0)
+  end
+
+  # KDB
+
+  def kdb_open(_contract, _parent_key) do
+    raise "kdb_open/2 not implemented"
```

Comment:
Who sees that error and when does it occur? Isn't it implemented in [src/bindings/erl_nif/lib/nif_kdb.c](https://github.com/ElektraInitiative/libelektra/pull/4623/files#diff-8e1b76f104e84f36aa85921e5cd0d1dc91f8c585effc73a9c44034397fa05398)? So is a better error maybe that the installation is broken or a system module is missing?

Reply:
This is a good point. With the current configuration this error should never occur.

If the module cannot be loaded this is an error in itself and would cause an exception to be raised before this one. (The failure would be in the `load_nifs` function a few lines above which gets called when the module is loaded for the first time.)

The idea behind the wording of the error message is that one might have different NIF libraries/implementations to choose from and not all of them need to implement every function.

In the present case this error message would only occur if there is a mismatch between the functions specified in the two files. I will adapt the message accordingly.